### PR TITLE
[ONNX] Remove static None graph output

### LINKED ISSUE #82370: [ONNX] Fail to export inception_v3 with torch script
### 🐛 Describe the bug

```python
import torch
import torchvision

x = torch.randn(1, 3, 299, 299)
model = torchvision.models.inception_v3()
model = torch.jit.script(model)
torch.onnx.export(model, x, 'model.onnx', verbose=True)
```

Error appears to be related to new onnx optional export.

### Versions

PyTorch version: 1.13.0a0+git795399c
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: 10.0.0-4ubuntu1
CMake version: version 3.22.2
Libc version: glibc-2.17

Python version: 3.7.11 (default, Jul 27 2021, 14:32:16)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.13.0-44-generic-x86_64-with-debian-bullseye-sid

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #82623

Fixes #82370 
* Unify the export behavior regarding static None outputs. These are
dropped for both traced graph and TorchScript graph export.
* `Optional` outputs are not affected.
Fixes #82370

### CODE REVIEW

**Review by titaiwangms — APPROVED** [review]
LGTM, CI is failing with onnx tests.

**Comment by BowenBao** [comment]
@abock for merge approval

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp [hunk 1]
```diff
@@ -2261,20 +2261,28 @@ size_t ONNXAssignOutputShape(
...
-    // Ideally we'd remove this difference.
-    outputs_index += static_cast<size_t>(is_script);
+    // in torch/jit/_trace.py while tracing. This means the traced IR graph
+    // has None outputs omitted.
+    // But then the outputs passed in here are un-flattened, which means they
+    // contain None objects. Ideally we'd remove this difference.
+    if (is_script && outputs_index < graph->outputs().size()) {
+      if (graph->outputs().at(outputs_index)->node()->mustBeNone()) {
+        graph->eraseOutput(outputs_index);
+      } else {
```
Comment (AllenTiTaiWang) [inline torch/csrc/jit/passes/onnx/shape_type_inference.cpp:2282]: Q: I am reading the related Optional type code, wondering is this else branch for Optional type output?

Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/shape_type_inference.cpp:2282]: Yes it is for optional type output. The comment above explains when and why this happens.

**Comment by BowenBao** [comment]
@abock please take a look

**Comment by BowenBao** [comment]
@abock for approval

**Review by abock — APPROVED** [review]
Sorry this one slipped through the cracks

**Pull request closed without merging**